### PR TITLE
Password script

### PR DESCRIPTION
**Pull request by jburel**
# What this PR does

Hide the password if used as a parameter
and use a secure connection to run the script if a password is specified

# Testing this PR

Run a script with a password

# Related reading

See https://github.com/openmicroscopy/openmicroscopy/pull/5598
see https://trello.com/c/C3EqebXJ/61-script-parameter-password

cc @will-moore

### CODE REVIEW

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/ScriptCallback.java [hunk 1]
```diff
@@ -74,12 +79,46 @@
...
+	private Gateway gw;
+	
+	/** Reference to the context. Only needed for encrypted connection.*/
+	private SecurityContext ctx;
+
+    /**
+     * Creates a new instance.
+     * 
+     * @param scriptID The identifier of the script to run.
+     * @param process  The process to handle.
```
Comment (mtbc) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/ScriptCallback.java]: should match arg name

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/ScriptCallback.java [hunk 2]
```diff
@@ -74,12 +79,46 @@
...
+	    results = null;
+	    this.gw = gw;
+	    this.ctx = ctx;
+	}
+
+	/**
+     * Creates a new instance.
+     * 
+     * @param scriptID The identifier of the script to run.
+     * @param process  The process to handle.
```
Comment (mtbc) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/ScriptCallback.java]: should match arg name

File: components/insight/SRC/org/openmicroscopy/shoola/agents/util/ui/ScriptingDialog.java [hunk 3]
```diff
@@ -490,13 +496,18 @@ public void mouseReleased(MouseEvent e) {
...
-                        ((JTextField) comp).setColumns(length);
-                        ((JTextField) comp).setText(s);
-                        columnsSet = s.length() > 0;
+                    String nl = name.toLowerCase();
+                    if ("password".equals(nl)) {
+                        comp = new JPasswordField();
+                    } else {
+                        comp = new JTextField();
+                        if (defValue != null) {
+                            length = defValue.toString().length();
```
Comment (mtbc) [inline components/insight/SRC/org/openmicroscopy/shoola/agents/util/ui/ScriptingDialog.java]: Definitely want the pre-trim length?

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java [hunk 4]
```diff
@@ -479,6 +480,50 @@ private long getScriptID(SecurityContext ctx, String name, String message)
 		return -1;
 	}
 
+	/**
```
Comment (mtbc) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: Weird indent.

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java [hunk 5]
```diff
@@ -479,6 +480,50 @@ private long getScriptID(SecurityContext ctx, String name, String message)
...
+        ScriptCallback cb = null;
+        try {
+            ProcessCallbackI pcb;
+            if (encrypted) {
+                Gateway g = new Gateway(dsFactory.getLogger());
+                UserCredentials uc = dsFactory.getCredentials();
+                LoginCredentials cred = new LoginCredentials();
+                cred.getUser().setUsername(uc.getUserName());
+                cred.getUser().setPassword(uc.getPassword());
+                cred.getServer().setHostname( uc.getHostName());
```
Comment (mtbc) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: remove space after `(`

File: components/insight/SRC/org/openmicroscopy/shoola/agents/util/ui/ScriptingDialog.java (hunk 3 above)
Comment (jburel) [reply components/insight/SRC/org/openmicroscopy/shoola/agents/util/ui/ScriptingDialog.java]: old code not changed in this PR. 
I could clean it

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java (hunk 4 above)
Comment (jburel) [reply components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: insight indent is a disaster i.e. mix of tabs and spaces

Comment (mtbc) [reply components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: Happy for these minor issues *not* to be a blocker given that Insight is maintenance-only anyway.

**Comment by jburel** [comment]
closing this investigation PR

**Pull request closed without merging**